File: src/main/java/com/itsci/mju/maebanjumpen/dto/HousekeeperDetailDTO.java
package com.itsci.mju.maebanjumpen.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true) // เพื่อให้เปรียบเทียบ Field ของ Parent Class ได้
public class HousekeeperDetailDTO extends HousekeeperDTO {
    // 🎯 ฟิลด์นี้มีเฉพาะใน Detail DTO เท่านั้น เพื่อตัดวงจร
    private List<HireDTO> hires;
    private int jobsCompleted; // 🎯 รับจำนวนงานที่เสร็จสิ้น
    private List<ReviewDTO> reviews; // 🎯 รับรายการรีวิว
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/housekeeperskill/dto/HousekeeperDetailDTO.kt
package com.itsci.mju.maebanjumpen.housekeeperskill.dto

import com.itsci.mju.maebanjumpen.hire.dto.HireDTO
import com.itsci.mju.maebanjumpen.partyrole.dto.HousekeeperDTO
import com.itsci.mju.maebanjumpen.review.dto.ReviewDTO

class HousekeeperDetailDTO : HousekeeperDTO() {
    var hires: List<HireDTO>? = null
    var jobsCompleted: Int = 0 // รับจำนวนงานที่เสร็จสิ้น
    var reviews: List<ReviewDTO>? = null // รับรายการรีวิว
}

